client/gutenberg/extensions/presets/jetpack/utils/refresh-registrations.js

```diff
@@ -0,0 +1,37 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+import { registerBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import getJetpackData from './get-jetpack-data';
+
+/**
+ * Refreshes registration of Gutenberg extensions (blocks and plugins)
+ *
+ * Uses block and plugin availability information obtained from the server to conditionally
+ * register and/or unregister blocks and plugins accordingly
+ *
+ * @returns {void}
+ */
+export default function refreshRegistrations() {
+	const extensions = getJetpackData();
+
+	if ( ! extensions ) {
+		return;
+	}
+
+	forEach( extensions, ( { available }, name ) => {
+		if ( available ) {
+			// TODO: Need settings. Probably export from individual block files
```

Comment:
>Probably export from individual block files

Nice, you'll find they're already exported in each `editor.js` exactly this use in mind:

https://github.com/Automattic/wp-calypso/blob/063fa1b3b1dea66620291372172994ed4ac04a03/client/gutenberg/extensions/publicize/editor.js#L29-L31